src/main/java/org/sonar/scm/git/blame/RepositoryBlameCommand.java

```diff
@@ -110,6 +113,15 @@ public RepositoryBlameCommand setFilePaths(@Nullable Set<String> filePaths) {
     return this;
   }
 
+  /**
+   * Add a callback to check the progress of the algorithm
+   * @param progressCallBack Consumer to be called each time a commit is processed by the algorithm
+   */
+  public RepositoryBlameCommand setProgressCallBack(Consumer<StatefulCommit> progressCallBack) {
```

Comment:
I wouldn't expose an internal object  used in the algorithm to the public interface.